subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/DetachedConfigurationsIntegrationTest.groovy

```diff
@@ -43,15 +41,12 @@ class DetachedConfigurationsIntegrationTest extends AbstractIntegrationSpec {
                 repositories {
                     maven { url "${mavenRepo.uri}" }
                 }
-                task checkDependencies {
-                    doLast {
-                        configurations.each { conf ->
-                            def declared = conf.dependencies
-                            def detached = project.configurations.detachedConfiguration(declared as Dependency[])
-                            def resolved = detached.resolvedConfiguration.getFirstLevelModuleDependencies()
-                            assert declared*.name == resolved*.moduleName
-                        }
-                    }
+
+                configurations.each { conf ->
+                    def declared = conf.dependencies
+                    def detached = project.configurations.detachedConfiguration(declared as Dependency[])
+                    def resolved = detached.incoming.resolutionResult.allComponents[0].dependencies // first item is root
```

Comment:
Why not use the `getRoot` method on `ResolutionResult`?

Applies elsewhere

Reply:
No good reason, will update.